### PR TITLE
dkpro-jwpl-hibernate-5.2.1-spring-4.3.0-migration

### PR DESCRIPTION
**Pull request by mawiesne**
provides changes that upgrade JWPL to use modern framework dependencies:
- Hibernate 5.2.1.Final (it was: 4.1.3.Final)
- Spring 4.3.1.RELEASE (it was: 3.1.1.RELEASE)
- applies necessary fixes that would break the code in WikiHibernateUtil
- removes deprecated warnings in several Hibernate bound classes related to Hibernate 4.x -> 5.x update
- updates some JDBC driver dependencies such as MySQL and PostgreSQL to more recent versions

### CODE REVIEW

**Comment by mawiesne** [comment]
Maybe it's worth releasing a '1.2.0' version to Maven central - in case the PR is acceptable for you.

Don't hesitate to comment/ask me in case of any question on my intention behind the proposed changes.

**Comment by rzo1** [comment]
+1 for this pull request!

File: de.tudarmstadt.ukp.wikipedia.api/src/test/resources/log4j.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Copyright 2016
 # Ubiquitous Knowledge Processing (UKP) Lab
-# Technische Universit�t Darmstadt
+# Technische Universit�t Darmstadt
```
Comment (daxenberger) [inline de.tudarmstadt.ukp.wikipedia.api/src/test/resources/log4j.properties]: was that intentionally?

**Comment by daxenberger** [comment]
Thanks a lot for the PR. We appreciate your interest and contributions very much.

GitHub is a great place for you to build your open source contributor profile and we love to take new contributors on board. We have the policy of asking for a contributor license agreement. As your proposed changes are substantial and maybe you plan to contribute even more, I would ask you to please could carefully review our contribution policy and contributor license agreement:

  http://dkpro.github.io/contributing/

If you are able to send me a signed copy, I'll be more than happy to include your contributions into the codebase!

File: de.tudarmstadt.ukp.wikipedia.api/src/test/resources/log4j.properties (hunk 1 above)
Comment (mawiesne) [inline de.tudarmstadt.ukp.wikipedia.api/src/test/resources/log4j.properties]: no, seems the IDE screwed that on up. I'll fix that.

Comment (reckart) [inline de.tudarmstadt.ukp.wikipedia.api/src/test/resources/log4j.properties]: Java properties files must be ISO 8859. Possibly the encoding in the repository was UTF-8 and the IDE tried to "fix" that.

**Comment by mawiesne** [comment]
@daxenberger - Just sent you a signed copy.

**Comment by daxenberger** [comment]
Got it; thanks!

**Pull request merged**